packs/chef/actions/lib/shellhelpers.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import sys
+import os
+import subprocess
+import select
+import io
+
+# Starts command in subprocess processing output as it appears.
+def shell_out(command, env={}):
```

Comment:
Defaulting argument value to a mutable object (list, dictionary) is usually a bad idea and will result in unexpected behavior - https://stackoverflow.com/questions/1132941/least-astonishment-in-python-the-mutable-default-argument

Everywhere you default an argument to an empty dict or list, you should default the argument to `None` and inside the function do `value = value or {}` or `value = value or []`.